hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/test/java/org/apache/hadoop/yarn/server/federation/store/records/TestFederationProtocolRecords.java

```diff
@@ -326,4 +333,71 @@ public void testGetReservationHomeSubClusterRequest() throws Exception {
     validatePBImplRecord(GetReservationHomeSubClusterRequestPBImpl.class,
         GetReservationHomeSubClusterRequestProto.class);
   }
+
+  @Test
+  public void testValidateApplicationHomeSubClusterEqual() throws Exception {
+    long now = Time.now();
+
+    ApplicationId appId1 = ApplicationId.newInstance(now, 1);
+    SubClusterId subClusterId1 = SubClusterId.newInstance("SC-1");
+    ApplicationHomeSubCluster applicationHomeSubCluster1 =
+        ApplicationHomeSubCluster.newInstance(appId1, subClusterId1);
+
+    ApplicationId appId2 = ApplicationId.newInstance(now, 1);
+    SubClusterId subClusterId2 = SubClusterId.newInstance("SC-1");
+    ApplicationHomeSubCluster applicationHomeSubCluster2 =
+        ApplicationHomeSubCluster.newInstance(appId2, subClusterId2);
+
+    assertEquals(applicationHomeSubCluster1, applicationHomeSubCluster2);
+  }
+
+  @Test
+  public void testValidateReservationHomeSubClusterEqual() throws Exception {
+    long now = Time.now();
+
+    ReservationId reservationId1 = ReservationId.newInstance(now, 1);
+    SubClusterId subClusterId1 = SubClusterId.newInstance("SC-1");
+    ReservationHomeSubCluster reservationHomeSubCluster1 =
+        ReservationHomeSubCluster.newInstance(reservationId1, subClusterId1);
+
+    ReservationId reservationId2 = ReservationId.newInstance(now, 1);
+    SubClusterId subClusterId2 = SubClusterId.newInstance("SC-1");
+    ReservationHomeSubCluster reservationHomeSubCluster2 =
+        ReservationHomeSubCluster.newInstance(reservationId2, subClusterId2);
+
+    assertEquals(reservationHomeSubCluster1, reservationHomeSubCluster2);
+  }
+
+  @Test
+  public void testSubClusterIdEqual() throws Exception {
+    SubClusterId subClusterId1 = SubClusterId.newInstance("SC-1");
+    SubClusterId subClusterId2 = SubClusterId.newInstance("SC-1");
+    assertEquals(subClusterId1, subClusterId2);
+  }
+
+  @Test
+  public void testSubClusterIdInfoEqual() throws Exception {
+    SubClusterIdInfo subClusterIdInfo1 = new SubClusterIdInfo("SC-1");
+    SubClusterIdInfo subClusterIdInfo2 = new SubClusterIdInfo("SC-1");
+    assertEquals(subClusterIdInfo1, subClusterIdInfo2);
+  }
+
+  @Test
+  public void testSubClusterPolicyConfiguration() throws Exception {
+
+    String queue1 = "queue1";
+    WeightedPolicyInfo policyInfo1 = mock(WeightedPolicyInfo.class);
+    ByteBuffer buf1 = policyInfo1.toByteBuffer();
+    SubClusterPolicyConfiguration configuration1 = SubClusterPolicyConfiguration
+        .newInstance(queue1, policyInfo1.getClass().getCanonicalName(), buf1);
+
+    String queue2 = "queue1";
+    WeightedPolicyInfo policyInfo2 = mock(WeightedPolicyInfo.class);
+    ByteBuffer buf2 = policyInfo1.toByteBuffer();
+    SubClusterPolicyConfiguration configuration2 = SubClusterPolicyConfiguration
+        .newInstance(queue2, policyInfo2.getClass().getCanonicalName(), buf2);
+
+    assertEquals(configuration1, configuration2);
+  }
+
 }
```

Comment:
No new line at end of file.

Reply:
Thank you very much for your help reviewing the code, I will fix it.